services/report/raw_upload_processor.py

```diff
@@ -164,6 +185,24 @@ def _adjust_sessions(original_report, to_merge_report, to_merge_session, current
         for f in flags_under_carryforward_rules
         if f not in to_partially_overwrite_flags
     ]
+    if upload is None and to_partially_overwrite_flags:
+        log.warning("Upload is None, but there are partial_overwrite_flags present")
+    if (
+        upload
+        and USE_LABEL_INDEX_IN_REPORT_PROCESSING_BY_REPO_SLUG.check_value(
+            repo_slug(upload.report.commit.repository), default=False
+        )
+        and to_partially_overwrite_flags
+    ):
+        label_index_service = LabelsIndexService(upload.report)
+        # TODO: Needs shared upload
+        # if original_report._labels_index is None:
+        #     label_index_service.set_label_idx(original_report)
+        # # Make sure that the labels in the reports are in a good state to merge them
+        # make_sure_orginal_report_is_using_label_ids(original_report)
```

Comment:
Where will these functions come from?

Reply:
they will be implemented in that file... check https://github.com/codecov/worker/pull/189/files#diff-a6e093dd207a55f7406f52e6aaaa6fc0533ac9b5cf965a0d8ffdffc257216f49